CONTRIBUTING.md

```diff
@@ -210,6 +210,16 @@ Abbreviations are in the header of the file which introduces them. We list here
   - L-algebra becomes `lalg`/`Lalg`
 - Partial order is abbreviated to `porder` or `POrder`, e.g., `porderType`, `CanPOrderMixin` in `order.v`
 
+#### Deprecations
+
+Definitions and lemmas should never be removed or renamed without warning users:
+they should be deprecated first, by at least one release.
+We use the pragma `deprecated` to implement deprecation of definitions and lemmas;
+see the many examples in the source code.
+We try to keep deprecation warnings for at least two years unless impossible
+(for example because of a name clash); after this period, deprecation warnings
```

Comment:
```suggestion
We try to keep deprecation warnings for at least two years unless we need to remove them
(for example because of a name clash, and in any case after at least one release); after this period, deprecation warnings
```